backend/dataall/modules/s3_datasets/db/dataset_table_repositories.py

```diff
@@ -39,6 +44,19 @@ def create_synced_table(session, dataset: S3Dataset, table: dict):
     def delete(session, table: DatasetTable):
         session.delete(table)
 
```

Comment:
nit/suggestion: we could move this to the data filter repository. There is already a paginated_data_filters which is basically doing list_data_filters + pagination

Reply:
good call out - moving